packages/probing-caller/src/services/callerService.ts

```diff
@@ -0,0 +1,76 @@
+import { logger } from "pagopa-interop-probing-commons";
+import { callerConstants } from "../utilities/constants.js";
+import {
+  responseStatus,
+  technology,
+  TelemetryDto,
+  EserviceContentDto,
+  TelemetryKoDto,
+  TelemetryOkDto,
+} from "pagopa-interop-probing-models";
+import { getKoReason } from "../utilities/errorMapper.js";
+import { ApiClientHandler } from "../utilities/apiClientHandler.js";
+import { KMSClientHandler } from "../utilities/kmsClientHandler.js";
+import { match } from "ts-pattern";
+import {
+  ApplicationError,
+  makeApplicationError,
+} from "../model/domain/errors.js";
+
+export const callerServiceBuilder = (
+  apiClientHandler: ApiClientHandler,
+  kmsClientHandler: KMSClientHandler
+) => {
+  return {
+    async performRequest(eservice: EserviceContentDto): Promise<TelemetryDto> {
+      const beforeRequestTimestamp: number = Date.now();
+
+      try {
+        const baseUrl = `${eservice.basePath[0]}${callerConstants.PROBING_ENDPOINT_SUFFIX}`;
+
+        logger.info(
+          `Perfoming Telemetry ${eservice.technology} request with eserviceRecordId: ${eservice.eserviceRecordId}`
+        );
+
+        const token: string = await kmsClientHandler.buildJWT(
+          eservice.audience
+        );
+
+        await match(eservice.technology)
+          .with(
+            technology.soap,
+            async () => await apiClientHandler.sendSOAP(baseUrl, token)
```

Comment:
we should get the `beforeRequestTimestamp` here, just before the call, otherwise we are including also the time required to generate our token

Reply:
beforeRequestTimestamp is needed also in the catch block, which is why it is initialized before try/catch

Reply:
in this case it's ok to use a `let` 👍